### PR TITLE
Updated windows window/frontend to fix issue with html select positioning

### LINKED ISSUE #1081: [v2] Select Dropdown doesn't update position when moving window
**Description**
When you open a select (dropdown) and move the window, the dropdown stays open at a fixed location. Also after moving the window with the dropdown closed, the dropdown first opens at the old location.

Note: For the workaround i had to make changes to this repo (v2) and also to the following repo:
github.com/leaanthony/go-webview2

**To Reproduce**
Create a new Wails app and insert a html select element. 
Also see this issue: https://github.com/MicrosoftEdge/WebView2Feedback/issues/780
And also this issue for a video of the problem: https://github.com/tauri-apps/tauri/issues/3089

**Expected behaviour**
Dropdown should close when it's open and you move the window (like it does in Chrome)
Dropdown should open at the correct position after moving the window

**Screenshots**
See video at: https://github.com/tauri-apps/tauri/issues/3089
Broken:
broken

Working
working

**System Details**
System
------
OS:             Windows 10 Pro N
Version:        2009 (Build: 19044)
ID:             21H2
Go Version:     go1.17.6
Platform:       windows
Architecture:   amd64

Dependency      Package Name    Status          Version
----------      ------------    ------          -------
WebView2        N/A             Installed       97.0.1072.55
npm             N/A             Installed       8.1.2
*upx            N/A             Available

* - Optional Dependency

**Additional context**
In the WndProc i called the w32.SetFocus(w.Handle()) function on the w32.WM_Moving message. This changes the focus on the webview and closes the dropdown when moving the window.

In the WndProc i called the NotifyParentWindowPositionChanged, this makes it so the webview will know it's position has changed and the dropdown opens in the correct place afterwards.
See this https://docs.microsoft.com/en-us/microsoft-edge/webview2/reference/win32/icorewebview2controller?view=webview2-1.0.774.44#notifyparentwindowpositionchanged

Do you want me to also open a PullRequest on the github.com/leaanthony/go-webview2 repo? I will warn you that the work around is fairly badly coded and was just to make sure i could actually fix the issue.

- [x] This issue is for Windows and I have read the [Windows Developer Guide](https://wails.app/guides/windows/)

### PR DESCRIPTION
**Pull request by MikeSchaap**
Fixes #1081

Need to wait for the following PR (https://github.com/leaanthony/go-webview2/pull/5) to merge so `go-webview2` exposes the `NotifyParentWindowPositionChanged()` method.

Was not sure how to correctly expose the chromium object to the window object. Do you agree with the proposed fix? Doesn't feel really solid imo

Note: First time creating a public Pull Request, so if I can improve certain parts, please let me know :)

### CODE REVIEW

**Comment by leaanthony** [comment]
I would suggest having the notify function in the Window struct rather than options, as that's user options. In frontend Run, you can set the notify callback directly on the window struct after setting up chromium

**Comment by MikeSchaap** [comment]
I moved the `notifyParentWindowPositionChanged` method to the Window struct and set the value in the `Run` method of `Frontend`. I was getting a nil pointer on the `WndProc` because it was called before the `notifyParentWindowPositionChanged` method was set. I moved the `winc.RegMsgHandler(w)` method call from the `NewWindow` method to the `Window.Run` method to make sure `notifyParentWindowPositionChanged` method was set before it was being used. Do you have any idea if this might interfere with anything else? From my (short) testing, everything seemed to still work.

**Comment by MikeSchaap** [comment]
Also, I guess we still have to wait for a new tag on the `go-webview2` repo before we can update the version in `go.mod` right?

File: v2/internal/frontend/desktop/windows/window.go [hunk 1]
```diff
@@ -86,5 +86,16 @@ func NewWindow(parent winc.Controller, appoptions *options.App) *Window {
 }
 
 func (w *Window) Run() int {
+	winc.RegMsgHandler(w)
```
Comment (leaanthony) [inline v2/internal/frontend/desktop/windows/window.go:89]: Could we put this back to where it was and deal with the nil pointer scenario?

File: v2/internal/frontend/desktop/windows/window.go [hunk 2]
```diff
@@ -86,5 +86,16 @@ func NewWindow(parent winc.Controller, appoptions *options.App) *Window {
...
 	return winc.RunMainLoop()
 }
+
+func (w *Window) WndProc(msg uint32, wparam, lparam uintptr) uintptr {
+	switch msg {
+	case w32.WM_NCLBUTTONDOWN:
+		w32.SetFocus(w.Handle())
+	case w32.WM_MOVE, w32.WM_MOVING:
+		w.notifyParentWindowPositionChanged()
```
Comment (leaanthony) [inline v2/internal/frontend/desktop/windows/window.go:98]:
```suggestion
		if w.notifyParentWindowPositionChanged != nil {
			w.notifyParentWindowPositionChanged()
		}
```

**Review by leaanthony — CHANGES_REQUESTED** [review]
I think this is almost ready to go 👍

**Comment by leaanthony** [comment]
Also need to resolve conflicts

**Comment by MikeSchaap** [comment]
Oh no, i'm pretty sure i messed up... I shouldn't work on sunday i guess.. Trying to figure out how to 'restore' this PR

**Comment by MikeSchaap** [comment]
Looks like I managed to revert the messup. Also moved the `winc.RegMsgHandler(result)` back to it's original position and resolved the conflicts. I believe we should be good to go now

**Comment by leaanthony** [comment]
> Also, I guess we still have to wait for a new tag on the `go-webview2` repo before we can update the version in `go.mod` right?

Nah, just update the dependency to use latest commit. I usually specify this like `go get .....@ab2c3b2`

**Comment by stffabi** [comment]
> > Also, I guess we still have to wait for a new tag on the `go-webview2` repo before we can update the version in `go.mod` right?
> 
> Nah, just update the dependency to use latest commit. I usually specify this like `go get .....@ab2c3b2` Do you need to update go.mod + go.sum in this PR?

PR #1105 already bumped the `go-webview2` dependency and that PR has been merged. Therefore we could merge this PR without needing to bump it. 
Should we squash the commits before merging?

**Comment by leaanthony** [comment]
LGTM!

**Comment by MikeSchaap** [comment]
Cheers 🍾

**Comment by leaanthony** [comment]
Congrats on your first PR!

**Pull request merged**